src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala

```diff
@@ -117,6 +134,40 @@ object VersionInjectorPlugin extends AutoPlugin {
         gitConfFile
     }
 
+  val injectCacheKeyTask =
+    injectCacheKey <<= (resourceManaged in Compile, organization, name, version, streams, gitMostRecentCommit) map { // scalastyle: ignore
+      (resourceManaged, org, name, version, s, gitMRC) =>
+        val cacheKeyConfFile = resourceManaged / org / cleanArtifactName(name) /
+    }
+
+  val stageChecksumTask = stageChecksum := {
+    import java.io.{ File => JFile }
+
+    def getFileList(dir: String): Seq[JFile] = new JFile(dir) match {
```

Comment:
You should look at [the sbt Paths APIs](http://www.scala-sbt.org/0.13/docs/Paths.html) - there are some really nice methods for doing this kind of thing.